lib/puppet/provider/nssdb/certutil.rb

```diff
@@ -0,0 +1,72 @@
+require 'fileutils'
+
+Puppet::Type.type(:nssdb).provide(:certutil) do
+  commands :certutil => 'certutil'
+  commands :openssl => 'openssl'
+  commands :mkdir => 'mkdir'
+
+  def create
+    create_nssdb_dir
+    generate_password_file
+    generate_nssdb
+  end
+
+  def destroy
+    destroy_password_file
+    destroy_nssdb
+  end
+
+  def exists?
+    nssdb_exists?
+  end
+
+  private
+
+  def create_nssdb_dir
+    mkdir('-p', resource[:nssdb_dir])
+  rescue Puppet::ExecutionFailure => e
+    Puppet.err("Failed to create NSS database directory at #{resource[:nssdb_dir]}: #{e}")
+    return nil
+  end
+
+  def generate_password_file
+    openssl(
+      'rand',
+      '-base64', '24',
+      '-out', resource[:password_file]
+    )
```

Comment:
Could we do this with Ruby instead of openssl?

Reply:
Second thought, with every other resource we have we generate a password or password file and then hand it to the resource (look at keystore for example). I think nssdb could (and should) be the same to not overload the type. 

If you look at my thought here (https://github.com/theforeman/puppet-certs/pull/344/files#r654092223) that is a way we could unify it such that NSS database works the same as keystore or truststore.